### ISSUE TITLE
Inconsistencies between the PR when it's viewed as checked out and not checked out

### ISSUE THREAD

**Issue by woppers** [Microsoft/vscode-pull-request-github#444]
Hi guys, Can anyone shed some light on this for me? Great tool by the way 👍 I can't wait to see how it evolves over the coming months 😄 

**TL;DR** 
The Problem: I have a branch which is outdated against master. In `Github` and `Github Pull Requests`, the branch seems to be being compared the version of master that the branch was cut from. But when I checkout the branch using `Github Pull Requests`, the branch seems to be being compared to the latest version of `master`. See below for a more in depth explanation and some screenshots.

**Detailed explanation:**
When I view a `PR` through the extension _without_ checking it out, the `PR` behaves the exact same as the `PR` does through Github.

I have pictures for this **not checked out** scenario:

1. You can see it's not checked out
2. I'm viewing the `PR` using the `Github Pull Requests` extension
3. Indicates the `PR` I'm looking at
4. Is verification that I don't have the PR checked out.

![image](https://user-images.githubusercontent.com/3513149/45413327-cc00d880-b670-11e8-8c6f-89ad79996ec1.png)

So here is the outdated code in the branch when viewed through `Github Pull Requests`:

1. I'm viewing the `PR` using the `Github Pull Requests` extension
2. Indicates I'm looking at the `layouts/default.tsx` file in the `PR` 
3. Is verification that I don't have the PR checked out.
4. Verifies the outdated code isn't showing up as change in the `PR`

![image](https://user-images.githubusercontent.com/3513149/45413824-fa32e800-b671-11e8-9f2e-46d25a2345cf.png)

And this image verifies that the `PR` on `Github` behaves the very same:

![image](https://user-images.githubusercontent.com/3513149/45414132-9b21a300-b672-11e8-8d4f-092da518ef56.png)

**So far, so good**

But when I check out the code:

1. You can see it is checked out
2. I'm viewing the `PR` as checked out using the `Github Pull Requests` extension
3. Is verification that I have the PR checked out.

![image](https://user-images.githubusercontent.com/3513149/45414348-079ca200-b673-11e8-8713-048b4c25cb66.png)

Then the outdated coded shows up as changes in the `PR` but I cannot comment on it:

1. I'm viewing the `PR` as checked out using the `Github Pull Requests` extension
2. Indicates I'm looking at the `layouts/default.tsx` file in the `PR`
3. Is verification that I don't have the PR checked out
4. Displays the outdated code in the branch
5. Illustrates the white line which indicates where I can comment

![image](https://user-images.githubusercontent.com/3513149/45414719-eab49e80-b673-11e8-9a5e-56d05a8dcba8.png)

I hope this has been useful/helpful. If you need any more information please let me know.

- Extension version: 
- VSCode Version: 1.27.1 (1.27.1)
- OS: macOS High Sierra - 10.13.6 (17G65)

Steps to Reproduce:

1. Open a PR on a branch which is outdated against master

**Comment by RMacfarlane** [comment]
@woppers Sorry, I missed this earlier!

There are two different code paths for how file changes are displayed before and after checkout, but they should show the same diff, so I think this is actually a bug.

Before check out, there is some chance that the changes don't exist locally - if you haven't fetched recently, you may not have the commits on disk to compare. So to handle this the git extension relies more heavily on GitHub, it fetches the change patches and attempts to reconstruct the files.

During the checkout, we switch to the PR branch and run fetch, so we assume we have the commit locally. We use `git show` with the file path and the head sha and base sha for the PR to get the file content for each of those commits and diff that content.

Just to clarify, by "outdated against master", you just mean some commits behind it?

**Comment by RMacfarlane** [comment]
This should be fixed as part of https://github.com/Microsoft/vscode-pull-request-github/commit/6d11839cbca6e35f1f2b6827122bbabc0e10a573, we were using a different commit for the base file in the "Changes in Pull Request" tree
